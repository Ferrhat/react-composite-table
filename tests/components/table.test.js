import React from 'react';
import {shallow} from 'enzyme';
import Table from '../../lib/components/table';

describe('Table', () => {
    const table = shallow(<Table columns={[]} data={[]} />);

    it('renders properly', () => {
        expect(table).toMatchSnapshot();
    });

    it('contains a message component', () => {
        expect(table.find('Message').length).toEqual(0);
        table.setState({isMessageActive: true});
        table.update();
        expect(table.find('Message').length).toEqual(1);
    });

    it('hides the message on click', () => {
        expect(table.state('isMessageActive')).toEqual(true);
        table.find('Message').simulate('click');
        expect(table.state('isMessageActive')).toEqual(false);
    });

});
