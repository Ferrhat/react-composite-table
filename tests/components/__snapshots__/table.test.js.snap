// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders properly 1`] = `
<div>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      >
        Confirmation
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <p>
        Do you want to delete the selected row with id: 
        -1
        ?
      </p>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}
      >
        No
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="defaultBlue"
        disabled={false}
        onClick={[Function]}
      >
        Yes
      </Button>
    </ModalFooter>
  </Modal>
  <Table
    bordered={false}
    bsClass="n-table"
    className="n-table-standard n-table-filter"
    condensed={false}
    hover={true}
    responsive={false}
    striped={true}
  >
    <thead>
      <tr>
        <th
          width="50"
        />
      </tr>
      <tr>
        <th />
      </tr>
    </thead>
    <tbody />
    <Footer
      defaultNumberOfRowsShow={10}
      onCurrentPageNumberChange={[Function]}
      onPageSizeChange={[Function]}
      tableData={Array []}
    />
  </Table>
</div>
`;
